irispy/spectrograph.py

```diff
@@ -3,300 +3,156 @@
-                data_dict = dict([(window_name, None) for window_name in self.spectral_windows["name"]])
-                # Record header info of auxiliary data.  Should be
-                # consistent between files of same OBS.
-                auxiliary_header = hdulist[-2].header
-            # Extract the data and meta/auxiliary data.
-            # Create WCS object from FITS header and add WCS object
-            # wcs dictionary.
-            wcs_celestial = wcs.WCS(hdulist[1].header).celestial
-            scan_label = "scan{0}".format(f)
-            wcs_celestial_objects[scan_label] = wcs_celestial
-            # Append to list representing the scan labels of each
-            # spectrum.
-            len_raster_axis = hdulist[1].header["NAXIS3"]
-            raster_index_to_file = raster_index_to_file+[scan_label]*len_raster_axis
-            # Append to list representing the raster positions of each
-            # spectrum.
-            raster_positions = raster_positions+list(range(len_raster_axis))
-            # Concatenate auxiliary data arrays from each file.
-            try:
-                auxiliary_data = np.concatenate((auxiliary_data, np.array(hdulist[-2].data)), axis=0)
-            except UnboundLocalError as e:
-                if e.args[0] == "local variable 'auxiliary_data' referenced before assignment":
-                    auxiliary_data = np.array(hdulist[-2].data)
-                else:
-                    raise e
-            # For each spectral window, concatenate data from each file.
+                # creating a empty list for every spectral window and each spectral window
+                # is a key for the dictionary.
+                data_dict = dict([(window_name, list())
+                                  for window_name in self.spectral_windows["name"]])
+
+            # the unchanged header of the hdulist indexed 0.
+            self.meta = hdulist[0].header
             for i, window_name in enumerate(self.spectral_windows["name"]):
-                # Set invalid data values to NaN.
-                data_nan_masked = copy.deepcopy(hdulist[window_fits_indices[i]].data)
-                data_nan_masked[hdulist[window_fits_indices[i]].data == -200.] = np.nan
+                wcs_ = WCS(hdulist[window_fits_indices[i]].header)
+                data_ = hdulist[window_fits_indices[i]].data
                 try:
-                    data_dict[window_name] = np.concatenate((data_dict[window_name], data_nan_masked))
+                    # appending Cube instance to the corresponding window key in dictionary's list.
+                    data_dict[window_name].append(Cube(data_, wcs_, meta=dict(self.meta)))
                 except ValueError as e:
-                    if e.args[0] == "zero-dimensional arrays cannot be concatenated":
-                        data_dict[window_name] = data_nan_masked
-                    else:
-                        raise e
-            # Close file.
+                    raise e
```

Comment:
If the error is just going to be raised with no additional output, is this except statement needed at all?  Probably best to just delete it.